src/js/entities-service/entities/patients.js

```diff
@@ -35,22 +35,27 @@ const _Model = BaseModel.extend({
   addGroup(group) {
     const groups = this.get('_groups') || [];
 
-    this.set('_groups', groups.concat(group.id));
+    this.set('_groups', groups.concat({
+      id: group.id,
+    }));
   },
-  removeGroup(group) {
+  removeGroup(removedGroup) {
     const groups = this.get('_groups');
 
-    this.set('_groups', reject(groups, groupId => groupId === group.id));
+    this.set('_groups', reject(groups, group => group.id === removedGroup.id));
   },
-  saveAll() {
-    const attrs = extend({}, this.attributes);
+  saveAll(attrs) {
+    attrs = omit(extend({}, this.attributes, attrs), 'engagement', 'source');
```

Comment:
This was an assumption that we would not want to include `engagement` and `source`. I could see leaving `source`, but engagement comes from a separate endpoint and is not treated as a true relationship.